### PR TITLE
[@sentry/webpack-plugin] Remove, bundles its own types

### PR DESCRIPTION
**Pull request by BendingBender**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: notNeededPackages.json [hunk 1]
```diff
@@ -3858,6 +3858,10 @@
             "libraryName": "sentence-case",
             "asOfVersion": "1.1.3"
         },
+        "sentry__webpack-plugin": {
+            "libraryName": "@sentry/webpack-plugin",
+            "asOfVersion": "1.10.0"
```
Comment (ImRodry) [inline notNeededPackages.json:3863]: This was added on commit https://github.com/getsentry/sentry-webpack-plugin/commit/b6b0e11e74fd55836d3299cef88531b2a34c2514#diff-7aa4473ede4abd9ec099e87fec67fd57afafaf39e05d493ab4533acc38547eb8 which is part of v1.9.0
1.9.0 changelog: https://github.com/getsentry/sentry-webpack-plugin/releases/tag/v1.9.0
```suggestion
            "asOfVersion": "1.9.0"
```

**Review by ImRodry — CHANGES_REQUESTED** [review]

File: notNeededPackages.json (hunk 1 above)
Comment (BendingBender) [reply notNeededPackages.json:3863]: Not possible, there is already a `@types/sentry__webpack-plugin@1.9.3` published on `npm`. I simply took the next version.

Comment (ImRodry) [reply notNeededPackages.json:3863]: This is supposed to be the number of the version that added type declarations for this package, it's not the version of the `@types` package

Comment (BendingBender) [reply notNeededPackages.json:3863]: The checks won't pass. You can try it out yourself with npm run test-all. This is actually supposed to be the version of the @types package that will be published with the obsoleted warning.

Comment (ImRodry) [reply notNeededPackages.json:3863]: Yeah you're right, my bad

**Review by ImRodry — APPROVED** [review]

**Pull request merged**